### PR TITLE
Fixes #35385 - Prevent auto-reset of password while editing

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

**Review by MariaAga — CHANGES_REQUESTED** [review]
Looks good, left one comment. There is a bug that if a user clicks on "change the password" icon, closes the modal (without saving) and opens it again (with the same or different webhook) the password is editable. 
I would set the value to false in the `onClose` function for the modal or something

**Comment by ofedoren** [comment]
Thanks, @MariaAga, updated. Should work now, but due to some limitations and just to not make a PR to the core, I needed to force users to use the buttons only to close the modal. Thus I've removed `x` mark in the header.

File: webpack/ForemanWebhooks/Routes/Webhooks/Components/WebhookForm/Components/ForemanFormikField.js [hunk 1]
```diff
@@ -56,10 +67,31 @@ const ForemanFormikField = ({
...
           );
           break;
+        case 'password':
+          content = setDisabled ? (
+            <div className="input-group">
+              {passwordInput}
+              <span className="input-group-btn">
+                <a
+                  className="btn btn-default"
+                  href="#"
```
[the remark this answers is not in the document]
Comment (MariaAga) [reply webpack/ForemanWebhooks/Routes/Webhooks/Components/WebhookForm/Components/ForemanFormikField.js]: @ofedoren any reasons for keeping it an `a` tag and not changing to button?

**Comment by MariaAga** [comment]
Yup, works well now!

File: webpack/ForemanWebhooks/Routes/Webhooks/Components/WebhookForm/Components/ForemanFormikField.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ofedoren) [reply webpack/ForemanWebhooks/Routes/Webhooks/Components/WebhookForm/Components/ForemanFormikField.js]: @MariaAga, my bad, forgot to change.

**Comment by ofedoren** [comment]
Updated with button instead of link.

**Review by MariaAga — APPROVED** [review]
lgtm!

**Comment by ofedoren** [comment]
Thanks, @MariaAga! Merging then :)

**Pull request merged**